### PR TITLE
add http client configurable timeout usefull when download is slow on server side

### PR DESCRIPTION
**Pull request by marian-craciunescu**
This Pull request fixes https://github.com/esp8266/Arduino/issues/1676 referenced on  https://github.com/openenergymonitor/EmonESP/issues/3 (fix stream error).

### CODE REVIEW

**Comment by marian-craciunescu** [comment]
@igrr @me-no-dev @d-a-v can you provide some feedback

**Comment by marian-craciunescu** [comment]
@d-a-v  I've put it to 30s  since the wifi router signal is quite bad at my place.

File: libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.cpp [hunk 1]
```diff
@@ -31,6 +31,12 @@ extern "C" uint32_t _SPIFFS_end;
 
 ESP8266HTTPUpdate::ESP8266HTTPUpdate(void)
 {
+    this->_http_client_timeout=8000;
+}
+
+ESP8266HTTPUpdate::ESP8266HTTPUpdate(int httpClientTimeout)
+{
+    this->_http_client_timeout = httpClientTimeout;
```
[the remark this answers is not in the document]
Comment (marian-craciunescu) [reply libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.cpp]: done.thank you

**Comment by marian-craciunescu** [comment]
@devyte  how can I trigger a REBUILD .The build failed but not from something I've modified.

File: libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.h [hunk 2]
```diff
@@ -103,6 +104,8 @@ class ESP8266HTTPUpdate
 
     int _lastError;
     bool _rebootOnUpdate = true;
+private:
+    int _http_client_timeout;
```
[the remark this answers is not in the document]
Comment (ToXIcGaming) [reply libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.h]: How is this used?

**Pull request merged**